src/handshake.rs

```diff
@@ -115,12 +126,27 @@ where
 				bytes.extend_from_slice(v.as_bytes())
 			}
 		}
-		if iter.peek().is_some() {
+		if extensions_iter.peek().is_some() {
 			bytes.extend_from_slice(b", ")
 		}
 	}
 }
 
+// This function takes a 16 byte key (base64 encoded, and so 24 bytes of input) that is expected via
+// the `Sec-WebSocket-Key` header during a websocket handshake, and a 32 byte output buffer, and
+// writes the response that's expected to be handed back in the response header `Sec-WebSocket-Accept`.
+//
+// See https://datatracker.ietf.org/doc/html/rfc6455#section-1.3 for more information on this.
+fn generate_accept_key<'k>(key_base64: &[u8; 24], output_buf: &'k mut [u8; 32]) -> &'k [u8] {
```

Comment:
FYI: there is a `WebSocketKey` type alias defined here (`[u8; 24]`): the hash is always 16 bytes, and base64 encoding of it is always 24 bytes. You could skip the `output_buf` and just return `WebSocketKey` by value here.

Reply:
Thanks for the pointer! I am now returning a 28 (not 32) byte array rather than taking in a buffer. Unless I'm mistaken (definitely possible :D), base64 encoding a 160bit hash will always use exactly 28 bytes!